app/src/main/java/com/tyranor/next/ui/game/GameShortcutCropDialog.kt

```
@@ -0,0 +1,367 @@
+import java.io.File
+import java.io.FileOutputStream
+import kotlin.math.max
+import kotlin.math.min
+import kotlin.math.ceil
+import kotlin.math.floor
+
+private const val CROP_SOURCE_MAX_DIMENSION_PX = 2048
+
+/** 自定义桌面快捷方式图标的正方形裁切器。 */
+@Composable
+internal fun GameShortcutCropDialog(
```

detekt (1.23.8)

[warning] The function GameShortcutCropDialog appears to be too complex based on Cognitive Complexity (complexity: 40). Defined complexity threshold for methods is set to '25'

(detekt.complexity.CognitiveComplexMethod)